napari/plugins/__init__.py

```diff
@@ -126,42 +113,13 @@ def register_function_widget(
             if isinstance(func, tuple):
                 msg += (
                     " To provide multiple function widgets please use "
-                    "a LIST of tuples"
+                    "a LIST of callables"
                 )
             warn(msg)
             continue
 
-        if not isinstance(magic_kwargs, dict):
-            warn(
-                f'Plugin {plugin_name!r} provided invalid type '
-                f'({type(magic_kwargs)}) for 2nd argument to {hook_name} '
-                f'for function {func.__name__!r}. Widget ignored.'
-            )
-            continue
-
-        valid_magic_kwargs = set(signature(func).parameters) | _magicgui_sig
-        extra_kwargs = set(magic_kwargs) - valid_magic_kwargs
-        if extra_kwargs:
-            warn(
-                f'Plugin {plugin_name!r} provided invalid magicgui kwargs '
-                f'to {hook_name} for function {func.__name__!r}: '
-                f'{extra_kwargs}. Valid kwargs are {valid_magic_kwargs}.'
-                'Widget ignored.'
-            )
-            continue
-
-        if not isinstance(dock_kwargs, dict):
-            warn(
-                f'Plugin {plugin_name!r} provided invalid type '
-                f'({type(dock_kwargs)}) for 3nd argument to {hook_name} '
-                f'for function {func.__name__!r}. Widget ignored.'
-            )
-            continue
-
         # Get function name
-        name = str(dock_kwargs.get('name', '')) or func.__name__.replace(
-            '_', ' '
-        )
+        name = func.__name__.replace('_', ' ')
```

Comment:
something I recently came across is that checking `callable(func)` is not the same as `isinstance(func, types.FunctionType)` ... and arbitrary class instances that have a `__call__` method (but not a `__name__`) will pass the first check throw an `AttributeError` here.  So we should either use `isinstance(func, types.FunctionType)` above, or use `name = getattr(func, '__name__', 'Unnamed Callable').replace('_', ' ')` here

Reply:
I'd probably vote  `isinstance(func, types.FunctionType)` and use `types.FunctionType` everywhere as that seems more in line with what we want. Thoughts?

Reply:
it's safer, we may be excluding valid clever uses for now, but we can open it up further if the need arises.